### ISSUE TITLE
When using tf.gradients in the forward pass, my codes stop at `compute_gradients`

### ISSUE THREAD

**Issue by liyinglr** [tensorflow/tensorflow#26686]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution: Linux Ubuntu 16.04
- TensorFlow installed from (source or binary): binary (conda install)
- TensorFlow version (use command below): 1.9.0
- Python version: 3.5
- CUDA/cuDNN version: 390.25
- GPU model and memory: GTX1080, 11GB

### 1. What I did:
My codes like this example: [tensorpack faster rcnn example](https://github.com/tensorpack/tensorpack/blob/f417c49fe45759fb2c69cafcabe6613ea85ec469/examples/FasterRCNN/train.py#L98)

I used DetectModel as the base class to implement the detection model of the SSD architecture. The training, optimizer, and get_inference_tensor_names are unchanged, the build_graph is reimplemented in the subclass, and the preprocess is modified. Other modules (such as anchor_generator, target_assign), I used the code in tensorflow offical models] (https://github.com/tensorflow/models/tree/master/research/object_detection).

my codes as follow
```

class MyMetaArch(ModelDesc):

    def inputs(self):
        ret = [
            tf.placeholder(tf.float32, (None, None, None, 1), 'image'),
            tf.placeholder(tf.float32, (8, None, 4), 'gt_boxes'),
            tf.placeholder(tf.int64, (8, None,), 'gt_labels')
        ]
        return ret

    def build_graph(self, *inputs):
        inputs = dict(zip(self.input_names, inputs))
        images = inputs["image"]

        x = Conv2D('conv1', images, filters=16, kernel_size=3, strides=(2, 2))
        x = Conv2D('conv2', x, filters=32, kernel_size=3, strides=(2, 2))
        x = Conv2D('conv3', x, filters=64, kernel_size=3, strides=(2, 2))
        x = Conv2D('conv4', x, filters=64, kernel_size=3, strides=(2, 2))
        ...
        class_prediction_with_background = tf.concat(...) # concat multi branches 
        class_prediction_with_background = class_prediction_with_background [:,:,1] # face detector, only care about the foreground

        gradients = tf.gradients(class_prediction_with_background , [images])[0]
        regular_vals = tf.reduce_mean(gradients)
        ...
        class_losses = class_losses + regular_vals
```
### 2. What I observed:
(1) **Include the ENTIRE logs here:**
But there are not any errors. The runtime log as follows， and and I added some extra logs.

```
[32m[0312 17:31:05 @interface.py:86][0m setup_graph
[32m[0312 17:31:05 @tower.py:217][0m _setup_input
[32m[0312 17:31:05 @input_source.py:220][0m Setting up the queue 'QueueInput/input_queue' with size Tensor("QueueInput/input_queue_Size:0", shape=(), dtype=int32) for CPU prefetching ...
[32m[0312 17:31:05 @tower.py:219][0m _setup_graph
[32m[0312 17:31:05 @trainers.py:186][0m SyncMultiGPUTrainerReplicated _setup_graph
[32m[0312 17:31:05 @trainers.py:189][0m _make_get_grad_fn
[32m[0312 17:31:05 @trainers.py:191][0m call_for_each_tower
[32m[0312 17:31:05 @training.py:109][0m Building graph for training tower 0 on device /gpu:0 ...
[32m[0312 17:31:05 @training.py:116][0m override_to_local_variable /gpu:0
[32m[0312 17:31:05 @tower.py:248][0m get_grad_fn: start
[32m[0312 17:31:05 @tower.py:280][0m NOT XLA_COMPILE: compute_grad_from_inputs
[32m[0312 17:31:05 @tower.py:253][0m compute_grad_from_inputs start
[32m[0312 17:31:05 @model_desc.py:262][0m start build_graph
[32m[0312 17:31:06 @registry.py:125][0m conv1/conv1 input: [8, 644, 644, 1]
[32m[0312 17:31:06 @registry.py:133][0m conv1/conv1 output: [8, 160, 160, 12]
...
[32m[0312 17:31:07 @registry.py:125][0m conv3_1 input: [8, 20, 20, 64]
[32m[0312 17:31:08 @registry.py:133][0m conv3_1 output: [8, 20, 20, 128]
[32m[0312 17:31:08 @registry.py:125][0m conv3_2 input: [8, 22, 22, 128]
[32m[0312 17:31:08 @registry.py:133][0m conv3_2 output: [8, 10, 10, 256]
[32m[0312 17:31:08 @registry.py:125][0m conv4_1 input: [8, 10, 10, 256]
[32m[0312 17:31:08 @registry.py:133][0m conv4_1 output: [8, 10, 10, 128]
[32m[0312 17:31:08 @registry.py:125][0m conv4_2 input: [8, 12, 12, 128]
[32m[0312 17:31:08 @registry.py:133][0m conv4_2 output: [8, 5, 5, 256]
[32m[0312 17:31:08 @registry.py:125][0m prediction_layers/box_encoding_predictor_0 input: [8, 20, 20, 64]
[32m[0312 17:31:08 @registry.py:133][0m prediction_layers/box_encoding_predictor_0 output: [8, 20, 20, 84]
[32m[0312 17:31:08 @registry.py:125][0m prediction_layers/class_predictor_0 input: [8, 20, 20, 64]
[32m[0312 17:31:08 @registry.py:133][0m prediction_layers/class_predictor_0 output: [8, 20, 20, 42]
[32m[0312 17:31:08 @registry.py:125][0m prediction_layers/box_encoding_predictor_1 input: [8, 10, 10, 256]
[32m[0312 17:31:08 @registry.py:133][0m prediction_layers/box_encoding_predictor_1 output: [8, 10, 10, 4]
[32m[0312 17:31:08 @registry.py:125][0m prediction_layers/class_predictor_1 input: [8, 10, 10, 256]
[32m[0312 17:31:08 @registry.py:133][0m prediction_layers/class_predictor_1 output: [8, 10, 10, 2]
[32m[0312 17:31:08 @registry.py:125][0m prediction_layers/box_encoding_predictor_2 input: [8, 5, 5, 256]
[32m[0312 17:31:08 @registry.py:133][0m prediction_layers/box_encoding_predictor_2 output: [8, 5, 5, 4]
[32m[0312 17:31:08 @registry.py:125][0m prediction_layers/class_predictor_2 input: [8, 5, 5, 256]
[32m[0312 17:31:08 @registry.py:133][0m prediction_layers/class_predictor_2 output: [8, 5, 5, 2]
[32m[0312 17:31:11 @FaceboxesModel.py:325][0m cls_losses: Tensor("tower0/Loss/Loss_1/mul:0", shape=(8, 8525), dtype=float32, device=/device:GPU:0)
[32m[0312 17:31:13 @FaceboxesModel.py:236][0m start regularize_cost
[32m[0312 17:31:13 @regularize.py:95][0m regularize_cost() found 35 variables to regularize.
[32m[0312 17:31:13 @regularize.py:20][0m The following tensors will be regularized: conv1/conv1/W:0,...,conv3_1/W:0, conv3_2/W:0, conv4_1/W:0, conv4_2/W:0,...
[32m[0312 17:31:13 @FaceboxesModel.py:239][0m end regularize_cost
[32m[0312 17:31:13 @FaceboxesModel.py:242][0m start add_n
[32m[0312 17:31:13 @FaceboxesModel.py:244][0m end add_n
[32m[0312 17:31:13 @model_desc.py:264][0m end build_graph
[32m[0312 17:31:13 @model_desc.py:268][0m _build_graph_get_cost return Tensor("tower0/total_cost:0", shape=(), dtype=float32, device=/device:GPU:0)
[32m[0312 17:31:13 @tower.py:255][0m get_cost_fn Tensor("tower0/total_cost:0", shape=(), dtype=float32, device=/device:GPU:0)
[32m[0312 17:31:13 @tower.py:268][0m get_opt_fn
```
The code seems to be stuck at this line: https://github.com/tensorpack/tensorpack/blob/master/tensorpack/train/tower.py#L261
The function is `compute_gradients`

**Comment by ppwwyyxx** [comment]
As I have commented in the [original issue](https://github.com/tensorpack/tensorpack/issues/1103#issuecomment-472099735), please try to provide enough information (code, instructions) so that others can __reproduce__ your issues. 
Otherwise it's unlikely your issues can be timely diagnosed. See also https://stackoverflow.com/help/mcve

**Comment by liyinglr** [comment]
@ppwwyyxx I think maybe caused by tf.concat. I try use the convolution before concat as a parameter of tf.gradients , the graph can be created.

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new/choose). Thanks!
